step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/TimeSeriesService.java

```diff
@@ -103,32 +100,46 @@ private TimeSeriesAggregationQuery mapToQuery(FetchBucketsRequest request) {
     @Produces(MediaType.APPLICATION_JSON)
     public AsyncTaskStatus<Object> rebuildTimeSeries(TimeSeriesRebuildRequest request) {
         String executionId = Objects.requireNonNull(request.getExecutionId(), "executionId not specified");
-        Equals measurementFilter = Filters.equals(MeasurementPlugin.ATTRIBUTE_EXECUTION_ID, executionId);
-        Measurement firstMeasurement = measurementCollection.find(measurementFilter,
-                new SearchOrder(MeasurementPlugin.BEGIN, 1), 0, 1, 0).findFirst().orElse(null);
-        Measurement lastMeasurement = measurementCollection.find(measurementFilter,
-                new SearchOrder(MeasurementPlugin.BEGIN, -1), 0, 1, 0).findFirst().orElse(null);
-        if (firstMeasurement != null && lastMeasurement != null) {
-            // Check if a time series already exist for this execution
-            TimeSeriesAggregationResponse aggregationResponse = aggregationPipeline.newQuery().range(firstMeasurement.getBegin(), lastMeasurement.getBegin())
-                    .filter(Map.of(MeasurementPlugin.ATTRIBUTE_EXECUTION_ID, executionId)).split(1l).run();
-            long seriesCount = aggregationResponse.getMatrix().size();
-            if (seriesCount > 0) {
-                throw new ControllerServiceException("Time series already exist for this execution. Unable to rebuild it");
-            } else {
+        if (this.timeSeriesExists(executionId)) {
+            throw new ControllerServiceException("Time series already exist for this execution. Unable to rebuild it");
+        } else {
+            // we need to check if measurements exists
+            Equals measurementFilter = Filters.equals(MeasurementPlugin.ATTRIBUTE_EXECUTION_ID, executionId);
+            Measurement firstMeasurement = measurementCollection.find(measurementFilter,
+                    new SearchOrder(MeasurementPlugin.BEGIN, 1), 0, 1, 0).findFirst().orElse(null);
+            Measurement lastMeasurement = measurementCollection.find(measurementFilter,
```

Comment:
Why do we need to check the existence of the lastMeasurement? is the presence of the first one not enough?